github/pulls.go

```diff
@@ -248,19 +248,29 @@ type PullRequestMergeResult struct {
 	Message *string `json:"message,omitempty"`
 }
 
+type PullRequestOptions struct {
+	Squash bool
+}
+
 type pullRequestMergeRequest struct {
 	CommitMessage *string `json:"commit_message"`
+	Squash        *bool   `json:"squash,omitempty"`
 }
 
 // Merge a pull request (Merge Button™).
 //
 // GitHub API docs: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-buttontrade
-func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string) (*PullRequestMergeResult, *Response, error) {
+func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string, options *PullRequestOptions) (*PullRequestMergeResult, *Response, error) {
 	u := fmt.Sprintf("repos/%v/%v/pulls/%d/merge", owner, repo, number)
 
-	req, err := s.client.NewRequest("PUT", u, &pullRequestMergeRequest{
-		CommitMessage: &commitMessage,
-	})
+	pullRequestBody := &pullRequestMergeRequest{CommitMessage: &commitMessage}
+	if options != nil {
+		pullRequestBody.Squash = &options.Squash
+	}
+	req, err := s.client.NewRequest("PUT", u, pullRequestBody)
+
+	// TODO: This header will be unnecessary when the API is no longer in preview.
+	req.Header.Add("Accept", mediaTypeSquashPreview)
```

Comment:
Looked a little more into it, and I can confirm it should be `Set("Accept", "mediaTypeSquashPreview")` here, not `Add`.

`mediaTypeSquashPreview` already includes the entire media type value:

```
application/vnd.github.polaris-preview+json
```

Which overrides the `v3` (the default version) with `github.polaris-preview` preview API version.

```
application/vnd.github.v3+json
```

Having both `Accept` headers wouldn't make sense.

```bash
$ curl https://api.github.com/users/hubot -I \
  -H "Accept: application/vnd.github.v3.full+json" \
  -H "Accept: application/vnd.github.polaris-preview+json" 2>/dev/null | grep X-GitHub-Media-Type
X-GitHub-Media-Type: github.v3; param=full; format=json

$ curl https://api.github.com/users/hubot -I \
  -H "Accept: application/vnd.github.polaris-preview+json" 2>/dev/null | grep X-GitHub-Media-Type
X-GitHub-Media-Type: github.polaris-preview; format=json
```

I'm not sure what one would do if some endpoint wanted to opt into two API previews at the same time. Maybe that never happens?